src/order/complete_well_founded.lean

```diff
@@ -109,9 +188,14 @@ lemma is_sup_closed_compact_iff_well_founded :
   is_sup_closed_compact α ↔ well_founded ((>) : α → α → Prop) :=
 (well_founded_characterisations α).out 2 0
 
+lemma is_Sup_finite_compact_iff_all_elements_compact :
+  is_Sup_finite_compact α ↔ ∀ k : α, is_compact_element k :=
+(well_founded_characterisations α).out 1 3
+
 alias well_founded_iff_is_Sup_finite_compact ↔ _ is_Sup_finite_compact.well_founded
 alias is_Sup_finite_compact_iff_is_sup_closed_compact ↔
       _ is_sup_closed_compact.is_Sup_finite_compact
 alias is_sup_closed_compact_iff_well_founded ↔ _ well_founded.is_sup_closed_compact
+alias is_Sup_finite_compact_iff_all_elements_compact ↔ _ all_elements_compact.is_Sup_finite_compact
```

Comment:
```suggestion
```
As it stands, this line creates an alias for the `←` direction of `is_Sup_finite_compact_iff_all_elements_compact` and calls it `all_elements_compact.is_Sup_finite_compact`. But `all_elements_compact.is_Sup_finite_compact` already exists so this line does nothing!

Now that we have 4 equivalent conditions, we have 12 possible implications between them. I wonder if we should only create aliases for `well_founded.foo` for the three values of `foo`?

Reply:
I'm a bit unsure what the utility of these aliases is. In any case where they were needed, one could simply use `{iff lemma}.(mp|mpr)`, no?

Reply:
It might be worth seeking @urkud 's opinion on the aliases.